GCBCore/Views/WidgetTemplateView.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  Created by Krzysztof Werys on 16.05.2016.
+//  Copyright © 2016 Oktawian Chojnacki. All rights reserved.
+//
+
+import UIKit
+
+/**
+ This viewModel is used to configure a template view for widgets: `WidgetTemplateView`
+ */
+public struct WidgetTemplateViewModel {
+    let title: String
+    let subtitle: String
+    let contentView: UIView
+
+    public init(title: String, subtitle: String, contentView: UIView) {
+        self.title = title
+        self.subtitle = subtitle
+        self.contentView = contentView
+    }
+}
+
+/**
+ Struct which is used to configure layout of a template view for widgets.
+ */
+public struct WidgetTemplateLayoutSettings {
+    let contentMargin: UIEdgeInsets
+    let displayContentUnderHeader: Bool
+
+    public init(contentMargin: UIEdgeInsets, displayContentUnderHeader: Bool = false) {
+        self.contentMargin = contentMargin
+        self.displayContentUnderHeader = displayContentUnderHeader
+    }
+}
+
+/**
+ This is template view for a widget which contains a header and a content.
+ */
+public class WidgetTemplateView: UIView {
+
+    @IBOutlet private weak var titleLabel: UILabel!
+    @IBOutlet private weak var descriptionLabel: UILabel!
+
+    @IBOutlet private weak var headerView: UIView!
+    @IBOutlet private weak var contentViewContainer: UIView!
+
+    @IBOutlet private weak var containerTopConstraint: NSLayoutConstraint!
+
+    public class func viewWithViewModel(viewModel: WidgetTemplateViewModel, layoutSettings: WidgetTemplateLayoutSettings) -> WidgetTemplateView {
+        let view = WidgetTemplateView.fromNib()
+        view.configureWithViewModel(viewModel)
+        view.configureLayoutSettings(layoutSettings)
+        return view
+    }
+
+    private func configureWithViewModel(viewModel: WidgetTemplateViewModel) {
+        configureContenView(viewModel.contentView)
+        configureTitle(viewModel.title)
+        configureSubtitle(viewModel.subtitle)
+    }
+
+    private func configureLayoutSettings(layoutSettings: WidgetTemplateLayoutSettings) {
+        if layoutSettings.displayContentUnderHeader {
+            containerTopConstraint.constant = 0
+        } else {
+            containerTopConstraint.constant = headerView.frame.height
```

Comment:
I am not sure if this is will be bulletproof in the future. Like, could we have two vertical constraints, one attached to the top of the view, and second attached to the bottom of the headerView?